INN/umbrella-energynewsnetwork
# Hardcode the Digest Sidebar, style fixes for certain widgets therein

## Changes

This pull request makes the following changes:

- Modifies ENN's `sidebar.php` to load a custom sidebar partial on the digest category page
- Adds a sidebar template to output the newly-created Digest Sidebar, or, if that is empty, nothing
- Registers a Digest Sidebar with appropriate description of where it is employed.

![Screen Shot 2020-02-19 at 16 37 27](https://user-images.githubusercontent.com/1754187/74880291-eb596600-5338-11ea-83ac-2fb13bcb2163.png)
![Screen Shot 2020-02-19 at 16 37 35](https://user-images.githubusercontent.com/1754187/74880292-ebf1fc80-5338-11ea-9733-678fe29d29c2.png)
![Screen Shot 2020-02-19 at 16 41 06](https://user-images.githubusercontent.com/1754187/74880293-ebf1fc80-5338-11ea-93c1-c21d4f247714.png)


## Why



For #86

## Testing/Questions

Features that this PR affects:

- /category/digest/


Questions that need to be answered before merging:

- [ ] do we want to do any specific styling for textwidget, with padding or anything like that? See https://github.com/INN/umbrella-energynewsnetwork/issues/86#issuecomment-588485946


Steps to test this PR:

1. Check out this branch
2. In Appearance > Widgets, add some widgets to the Digest Sidebar widget area. I added:
    - a text widget identifying the sidebar
    - a ENN Mailchimp signup widget
    - a textwidget that contains the following:
        ```html
        <p>
        	Are you subscribed to one or more daily digests? Add or remove subscriptions by following this link:
        </p>
        
        <a class="btn btn-primary" href="https://example.org">Manage Subscription</a>
        ```
3. Visit the following locations:
    - /category/digest/ : the Digest sidebar should be displayed
    - /category/news/ : the Digest sidebar should not be displayed
    - /region/west/ : the Digest sidebar should not be displayed
    - /category/digest/ with a search query: the Digest sidebar should be displayed
